helper/semvercon/semvercon.go

```diff
@@ -0,0 +1,147 @@
+// semvercon is a Semver Constraints package copied from
```

Comment:
`helper` seems like an odd root package. Maybe `lib`? If you wanted to set it up to eventually move the constraints implementations out of structs, I would suggest `lib/constraint/semver`.

Will let you decide what to do here.

Reply:
A quick glance at other Hashitools shows:
* TF, Packer, and Vault use `helper/`
* Consul uses `lib/`
* Nomad uses both :man_facepalming: 

Since `helper/` has more history, more use, and follows the style of more HashiCorp tools, I'm going to stick with it.

I'll rename to `helper/constraint/semver` though.